### PR TITLE
Blog app: add API endpoints

### PR DESCRIPTION
**Pull request by ElsonOtake**
- An API endpoint to list all posts for a user was created;
- An API endpoint to list all comments for a user's post was created;
- An API endpoint to add a comment to a post was created. The owner of the comment is the user that is logged in;
- Authentication to API endpoints was added;
- The API endpoints receive JSON and respond JSON as well.

### CODE REVIEW

File: app/controllers/authentication_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  ALLOWED_DATA = %(email password).freeze
+
+  # POST /auth/login
+  def login
+    data = json_payload.select { |allow| ALLOWED_DATA.include?(allow) }
+    @user = User.find_by_email(data[:email])
+    if @user.valid_password?(data[:password])
+      token = JsonWebToken.encode(user_id: @user.id)
+      time = Time.now + 24.hours.to_i
```
Comment (uche-inyama) [inline app/controllers/authentication_controller.rb:12]: I had an **(uninitialized constant JsonWebToken::JWT)**  error that pointed to line 11, when I tried to login.

**Review by uche-inyama — CHANGES_REQUESTED** [review]
### Hi @ElsonOtake, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/authentication_controller.rb (hunk 1 above)
Comment (ElsonOtake) [reply app/controllers/authentication_controller.rb:12]: Sorry for that. I forgot to add tests for unregistered emails.

**Review by uche-inyama — APPROVED** [review]
### Hi @team, 

Your project is complete!  Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**